cf-net/cf-net.c

```diff
@@ -0,0 +1,507 @@
+{
+    extern char *optarg;
+    extern int optind;//, opterr, optopt;
+    if (opts == NULL)
+    {
+        return -1;
+    }
+    CFNetSetDefault(opts);
+    int c = 0;
+    int start_index = 1;
+    while ((c = getopt_long(    argc, argv, "hMdvInVilb:t:",
+                                OPTIONS, &start_index))
+            != -1)
+    {
+        switch (c)
+        {
+            case 'h':
+            {
+                Writer *w = FileWriter(stdout);
+                WriterWriteHelp(w, "cf-net", OPTIONS, HINTS, true);
+                FileWriterDetach(w);
+                exit(EXIT_SUCCESS);
+                break;
+            }
+            case 'd':
+            {
+                opts->debug = true;
+                break;
+            }
+            case 'v':
+            {
+                opts->verbose = true;
+                break;
+            }
+            case 'I':
+            {
+                opts->inform = true;
+                break;
+            }
+            case 'n':
+            {
+                opts->dry_run = true; // TODO: Not in use
+                break;
+            }
+            case 'V':
+            {
+                printf("cf-net version number "CF_NET_VERSION"\n");
+                exit(EXIT_SUCCESS);
+                break;
+            }
+            case 'i':
+            {
+                opts->interactive = true; // TODO: Not in use
+                break;
+            }
+            case 'l':
+            {
+                opts->timestamp = true; // TODO: Not in use
+                break;
+            }
+            case 'b':
+            {
+                opts->background = atoi(optarg); // TODO: Not in use
+                break;
+            }
+            case 't':
+            {
+                opts->timeout = atoi(optarg); // TODO: Not in use
+                break;
+            }
+            case 'M':
+            {
+                // TODO: How do we actually add a man page
+                Writer *out = FileWriter(stdout);
+                ManPageWrite(out, "cf-net", time(NULL),
+                             CF_NET_SHORT_DESCRIPTION,
+                             CF_NET_MANPAGE_LONG_DESCRIPTION,
+                             OPTIONS, HINTS,
+                             true);
+                FileWriterDetach(out);
+                exit(EXIT_SUCCESS);
+                break;
+            }
+            default:
+            {
+                break;
+            }
+        }
+    }
+    (*cmd) = argv[optind];
+    (*args) = &(argv[optind+1]);
+    return 0;
+}
+
+// Just for testing argument parsing
+int CFNetPrint(char **args)
+{
+    for (int i = 0; args[i] != NULL; ++i)
+    {
+        printf("%s ", args[i]);
+    }
+    printf("\n");
+    return 0;
+}
+
+void CFNetInit()
+{
+    CryptoInitialize();
+    LoadSecretKeys();
+    cfnet_init(NULL, NULL);
+}
+
+AgentConnection* CFNetOpenConnection(char *server)
+{
+
+    AgentConnection *conn = NULL;
+    {
```

Comment:
To isolate the variables only used there. I think this used to be part of a longer function. Fixed.